src/bit-systems/link-hover-menu.ts

```diff
@@ -0,0 +1,51 @@
+import { defineQuery, enterQuery, entityExists } from "bitecs";
+import type { HubsWorld } from "../app";
+import { Interacted, Link, LinkHoverMenu, LinkHoverMenuItem, HoveredRemoteRight } from "../bit-components";
+import { anyEntityWith } from "../utils/bit-utils";
+
+const NULL_EID = 0;
+
+const hoveredQuery = defineQuery([HoveredRemoteRight, Link]);
+const hoveredEnterQuery = enterQuery(hoveredQuery);
+const hoveredMenuItemQuery = defineQuery([HoveredRemoteRight, LinkHoverMenuItem]);
+const clickedMenuItemQuery = defineQuery([Interacted, LinkHoverMenuItem]);
+
+export function linkHoverMenuSystem(world: HubsWorld) {
+  // Assumes always only single LinkHoverMenu entity exists.
+  const menuEid = anyEntityWith(world, LinkHoverMenu)!;
+  const menuObject = world.eid2obj.get(menuEid)!;
+
+  // Save Link object eid in LinkHoverMenu when hovered.
+  hoveredEnterQuery(world).forEach((eid: number) => {
+    LinkHoverMenu.targetObject[menuEid] = eid;
+    const targetObject = world.eid2obj.get(eid)!;
+    targetObject.add(menuObject);
+  });
+
+  // Check if the cursor it hovered on Link object or Link menu button object.
+  const hovered = hoveredQuery(world).length > 0 || hoveredMenuItemQuery(world).length > 0;
+
+  if (hovered) {
+    // Hovered then make the menu visible and handle clicks if needed.
+    menuObject.visible = true;
+    if (clickedMenuItemQuery(world).length > 0) {
+      const linkEid = LinkHoverMenu.targetObject[menuEid];
+      if (linkEid !== NULL_EID) {
+        // Just in case
+        if (entityExists(world, linkEid)) {
+          window.open(APP.getString(Link.url[linkEid])!);
```

Comment:
This accounts for the simple case of an external url but the current link component also supports linking to anchors in the same room, switching scenes, and switching avatars. See https://github.com/mozilla/hubs/blob/aaa4584a822545a8ea998c3c14edc62078fb542b/src/components/open-media-button.js#L43-L75 - Those should ideally also be reflected in the label.

Reply:
Do you think it's ok to support them in another PR in this or later cycle because I'm not really sure how difficult they are yet. I hope I can push the very basic function to the repository for now.

Reply:
I will add TODO comment for now